internal/fdasd/fdasd.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +build s390x
+
+package fdasd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/log"
+)
+
+type Operation struct {
+	logger *log.Logger
+	dev    string
+	config string
+}
+
+// Begin begins an fdasd operation
+func Begin(logger *log.Logger, dev string) *Operation {
+	os.Remove("/tmp/part-dasd")
+	return &Operation{logger: logger, dev: dev, config: "/tmp/part-dasd"}
+}
+
+// CreatePartition adds the supplied partition start and size to a file which is used by fdasd to create
+// partitions in non interactive mode. The non interactive options for fdasd are very limited and it uses
+// a very simple input file to configure partitions. Specified here: https://www.ibm.com/support/knowledgecenter/linuxonibm/com.ibm.linux.z.lgdd/lgdd_r_fasdusingoptions.html
+// a sample fdasd configuration file to configure partitions looks like this:
+//    [2,500]
+//    [501,1100]
+//    [1101,45000]
+// where the values are 'tracks' and 2 is the lowest track number to begin with. These partitions always have to be in
+// ascending order of values and cannot overlap. Violation of these conditions will result in fdasd rejecting the config.
+// The 'native' keyword used  below refers to the type of the partition and can be one of many like swap, raid etc..but
+// it is just a guideline.
+// One more thing to note is there is no way to preserve existing partitions. When a config is provided, fdasd destroys other partitions
+// and creates the partitions anew.
+func (op *Operation) CreatePartition(p types.Partition) error {
```

Comment:
`fdasd` silently ignores all partitions after the first three.  We should keep track of the number of partitions and fail if we receive too many.

Reply:
ah..yes this is another check i have been meaning to add. thanks.